pkg/controller/infrastructure/infraflow/reconcile.go

```diff
@@ -967,22 +1064,67 @@ func (c *FlowContext) deleteNATGateway(zoneName string) flow.TaskFn {
 	}
 }
 
+func (c *FlowContext) ensureEgressOnlyInternetGateway(ctx context.Context) error {
+	if isIPv6(c.ipFamilies) {
+
+		log := c.LogFromContext(ctx)
+		desired := &awsclient.EgressOnlyInternetGateway{
+			Tags:  c.commonTags,
+			VpcId: c.state.Get(IdentifierVPC),
+		}
+		current, err := findExisting(ctx, c.state.Get(IdentifierEgressOnlyInternetGateway), c.commonTags,
+			c.client.GetEgressOnlyInternetGateway, c.client.FindEgressOnlyInternetGatewaysByTags)
+		if err != nil {
+			return err
+		}
+
+		if current != nil {
+			c.state.Set(IdentifierEgressOnlyInternetGateway, current.EgressOnlyInternetGatewayId)
+			if _, err := c.updater.UpdateEC2Tags(ctx, current.EgressOnlyInternetGatewayId, c.commonTags, current.Tags); err != nil {
+				return err
+			}
+		} else {
+			log.Info("creating...")
+			created, err := c.client.CreateEgressOnlyInternetGateway(ctx, desired)
+			if err != nil {
+				return err
+			}
+			c.state.Set(IdentifierEgressOnlyInternetGateway, created.EgressOnlyInternetGatewayId)
+		}
+	}
+	return nil
```

Comment:
- reverse the conditional for an early return
- can the ip families change in a running shoot ? because if yes, then maybe a cleanup is also needed here

Reply:
I changed the condition here. 
For now ipfamilies are immutable. However, later we will want to migrate ipv4 shoots to dualstack and also back. However, I would add the clean-up code later, when we need it and are able to test it.